data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/DefaultCosmosRepositoryOperations.java

```diff
@@ -390,80 +497,110 @@ private String getId(RuntimePersistentEntity<?> persistentEntity, ObjectNode ite
+                    }
                     return value;
                 }
-                AttributeConverter<Object, Object> converter = attributeConverterRegistry.getConverter(converterClass);
-                ConversionContext conversionContext = createTypeConversionContext(null, argument);
-                return converter.convertToPersistedValue(value, conversionContext);
-            }
 
-            private ConversionContext createTypeConversionContext(@Nullable RuntimePersistentProperty<?> property,
-                                                                  @Nullable Argument<?> argument) {
-                if (property != null) {
-                    return ConversionContext.of(property.getArgument());
+                @Override
+                public Object convert(Class<?> converterClass, Object value, Argument<?> argument) {
+                    if (converterClass == null) {
+                        return value;
+                    }
+                    AttributeConverter<Object, Object> converter = attributeConverterRegistry.getConverter(converterClass);
+                    ConversionContext conversionContext = createTypeConversionContext(null, argument);
+                    return converter.convertToPersistedValue(value, conversionContext);
                 }
-                if (argument != null) {
-                    return ConversionContext.of(argument);
+
+                private ConversionContext createTypeConversionContext(@Nullable RuntimePersistentProperty<?> property,
+                                                                      @Nullable Argument<?> argument) {
+                    if (property != null) {
+                        return ConversionContext.of(property.getArgument());
+                    }
+                    if (argument != null) {
+                        return ConversionContext.of(argument);
+                    }
+                    return ConversionContext.DEFAULT;
                 }
-                return ConversionContext.DEFAULT;
-            }
 
-            @Override
-            public void bindOne(QueryParameterBinding binding, Object value) {
-                paramList.add(new SqlParameter("@" + binding.getRequiredName(), value));
-            }
+                @Override
+                public void bindOne(QueryParameterBinding binding, Object value) {
+                    boolean updateParameter = false;
+                    if (updateQuery) {
+                        String[] propertyPath = binding.getRequiredPropertyPath();
+                        PersistentPropertyPath pp = persistentEntity.getPropertyPath(propertyPath);
+                        if (pp != null) {
+                            String propertyName = pp.getPath();
+                            if (CollectionUtils.isNotEmpty(updatingProperties) && updatingProperties.contains(propertyName)) {
```

Comment:
What if I have: `update name = :q1 where name = q2`

Reply:
Current implementation relies that we build queries and don't support raw queries for update and delete, because these don't really exist in Azure Cosmos.
If I misunderstood can you please clarify question and use case.

Reply:
The logic there looks like will collect the property which is updating, but in some cases you can also query by the property which is being updated

Reply:
Ok, I got what you are saying. No need to skip adding params to the parameter list even if it is for update as binder will bind what it needs in the query. Fixed it, tnx